### ISSUE TITLE
Pass arguments to adapters as object

### ISSUE THREAD

**Issue by probil** [kuitos/axios-extensions#18]
It might be better to pass arguments to adapters as an object.
I mean, instead of:
```js
cacheAdapterEnhancer(adapter, false, 'cache', new LRUCache({ maxAge: FIVE_MINUTES }))
```
use
```js
const options = { cacheEnabledByDefault: false, enableCacheFlag: 'cache', defaultCache: new LRUCache({ maxAge: FIVE_MINUTES }) };
cacheAdapterEnhancer(adapter, options)
```

### Advantages:
**1) easier to use default values of given params**
For example, currently if I want to pass only custom cache I have to define all other arguments as `undefined`. And it looks weird:
```js
const cache =  new LRUCache({ maxAge: 1000 * 60 * 10 })
cacheAdapterEnhancer(axios.defaults.adapter, undefined, undefined, cache);
// compare with
cacheAdapterEnhancer(axios.defaults.adapter, { defaultCache: cache });
```

**2) more explicit (self-descriptive) arguments**
```js
cacheAdapterEnhancer(axios.defaults.adapter, false); // <-- what's false here?
// compare with
cacheAdapterEnhancer(axios.defaults.adapter, { cacheEnabledByDefault: false } );
```
It's hard to recognize what is `false` without checking the docs. It is not good developer experience

What do you think about that @kuitos ?

**Comment by kuitos** [comment]
Yes a object definition may be better in current scenario, thanks for your advice!
And, would you like to create a pr to improve it? 😀

**Comment by kuitos** [comment]
argument changed and released v3.0.0

**Comment by probil** [comment]
Awesome!
